### PR TITLE
Power Effect with SignalProducer from ReactiveSwift

### PR DESCRIPTION
**Pull request by haaakon**
Throwing my hat in the ring on refactoring the project to use another `Reactive` Framework. I've tested the app and it seems working.

Though my implementation does not provide the `.sync` function which i'll have to look more into. I started a discussion on how much of `Combine` functionality a solution like this should replace. In a separate project i've tried removing everything that makes the app support iOS versions lower than 13, which is a bigger undertaking, since you can't use the `@Published` property wrapper in `Store`, it has to be replaced with a `MutableProperty<Value>`.   

Thanks for doing this awesome series, i've learned alot and can't wait to start using this style more in projects moving forward!

### CODE REVIEW

File: 0081-combine-and-effects-pt2/ReactiveSwiftPrimeTime/Counter/WolframAlpha.swift [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    return Effect<Data?>.init { observer, lifetime in
+        let task = URLSession.shared.dataTask(with: components.url!) { data, _, _ in
+            observer.send(value: data)
+            observer.sendCompleted()
+        }
+        task.resume()
+    }
+    .skipNil()
+    .filterMap { try? JSONDecoder.init().decode(WolframAlphaResult.self, from: $0) }
+    .map { WolframAlphaResult.init(queryresult: $0.queryresult)}
```
Comment (mbrandonw) [inline 0081-combine-and-effects-pt2/ReactiveSwiftPrimeTime/Counter/WolframAlpha.swift]: would it also work to remove this `map`? It seems to extract out `queryresult` and then wrap it back up in `WolframAlphaResult`.

File: 0081-combine-and-effects-pt2/ReactiveSwiftPrimeTime/FavoritePrimes/FavoritePrimes.swift [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+private let loadEffect = Effect<FavoritePrimesAction?> { obs, _ in
+  let documentsPath = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)[0]
+  let documentsUrl = URL(fileURLWithPath: documentsPath)
+  let favoritePrimesUrl = documentsUrl.appendingPathComponent("favorite-primes.json")
+  guard
+    let data = try? Data(contentsOf: favoritePrimesUrl),
+    let favoritePrimes = try? JSONDecoder().decode([Int].self, from: data)
+    else { return  }
+    obs.send(value: .loadedFavoritePrimes(favoritePrimes))
+    obs.sendCompleted()
```
Comment (mbrandonw) [inline 0081-combine-and-effects-pt2/ReactiveSwiftPrimeTime/FavoritePrimes/FavoritePrimes.swift]: I believe you could create a `sync` that simply works `work` and then immediately completes. That way the ReactiveSwift would match exactly the API we have for combine (and the other Rx branch). Could you give that a shot?

File: 0081-combine-and-effects-pt2/ReactiveSwiftPrimeTime/Counter/WolframAlpha.swift (hunk 1 above)
Comment (haaakon) [reply 0081-combine-and-effects-pt2/ReactiveSwiftPrimeTime/Counter/WolframAlpha.swift]: Yeah, thats a small blunder, fixed

File: 0081-combine-and-effects-pt2/ReactiveSwiftPrimeTime/FavoritePrimes/FavoritePrimes.swift (hunk 2 above)
Comment (haaakon) [reply 0081-combine-and-effects-pt2/ReactiveSwiftPrimeTime/FavoritePrimes/FavoritePrimes.swift]: Yeah, not that hard, fixed.

**Comment by haaakon** [comment]
Thanks for the feedback, i puushed a commit with your requested changes. 

I would be very interested to see how different the architecture would be without needing to interface with SwiftUI, so consider that a vote for you guys covering that in the future :)

**Comment by fonkadelic** [comment]
It's cool to see how the effects work with ReactiveSwift.
One thing i noticed here is that the memory management that was necessary for the Combine implementation (cleanup cancelables on complete) is missing here. 
Does ReactiveSwift take care of this automatically? I'm still trying to understand the rules of ReactiveSwifts memory management. Would be great if somebody has input.
/cc @stephencelis

**Pull request merged**